open-reac/src/main/java/com/powsybl/openreac/parameters/input/AbstractElementsInput.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openreac.parameters.input;
+
+import com.powsybl.ampl.converter.AmplSubset;
+import com.powsybl.ampl.executor.AmplInputFile;
+import com.powsybl.commons.util.StringToIntMapper;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * @author Nicolas Pierre <nicolas.pierre at artelys.com>
+ */
+public abstract class AbstractElementsInput implements AmplInputFile {
+
+    private final List<String> elementIds;
+
+    protected AbstractElementsInput(List<String> elementIds) {
+        this.elementIds = elementIds;
+    }
+
+    @Override
+    public InputStream getParameterFileAsStream(StringToIntMapper<AmplSubset> stringToIntMapper) {
```

Comment:
I think we have an API issue here as to provider extra input file we have to provide an `InputStream` and looking at next line of code we need to load the file into an String and create an `InputStream` from the String.
Maybe we should have instead an OutputStream passed to the method so that we can write directly the file content.